packages/@react-spectrum/overlays/src/Popover.tsx

```diff
@@ -102,6 +102,49 @@ const PopoverWrapper = forwardRef((props: PopoverWrapperProps, ref: RefObject<HT
   let {modalProps} = useModal({
     isDisabled: isNonModal
   });
+  let [isHideArrow, setIsHideArrow] = useState(hideArrow);
+
+  useLayoutEffect(() => {
+    // check position of arrow and popover proximety to boundry edge
+    if (ref.current && arrowProps) {
+      let propsHideArrow = hideArrow;
+
+      // figure out the location of the offset and which popover dimension to use
+      let offset = arrowProps.style.left ? 'left' : 'top';
+      let overlayCrossSize = offset === 'left' ? ref.current.offsetWidth : ref.current.offsetHeight;
+
+      // for the start and top, no change is made to arrowPosition not caught in a condiational below
+      if (DEFAULT_MODAL_PADDING > arrowProps.style[offset] && arrowProps.style[offset] > 0) {
+        // using the default padding for proper arrow placement
+        let arrowStyle = {...arrowProps.style};
+        arrowStyle[offset] = DEFAULT_MODAL_PADDING;
+        arrowProps.style = arrowStyle;
```

Comment:
why make a copy and then overwrite it?
```suggestion
        arrowProps.style[offset] = DEFAULT_MODAL_PADDING;
```

Reply:
Doing the suggested change doesn't update the value. The above code was the solution I found that updates the value. It's been too long since I wrote this for me to remember the reason, sorry.